crates/sui-deepbook-indexer/src/server.rs

```diff
@@ -61,3 +82,59 @@ async fn get_pools(
 
     Ok(Json(results))
 }
+
+async fn get_24hr_volume(
+    Path(pool_id): Path<String>,
+    State(state): State<PgDeepbookPersistent>,
+) -> Result<Json<i64>, DeepBookError> {
+    let connection = &mut state.pool.get().await?;
+    let unix_ts = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as i64;
+    let day_ago = unix_ts - 24 * 60 * 60 * 1000;
+    let results: Vec<OrderFill> = schema::order_fills::table
+        .select(OrderFill::as_select())
+        .filter(schema::order_fills::pool_id.eq(pool_id))
+        .filter(schema::order_fills::onchain_timestamp.gt(day_ago))
+        .load(connection)
+        .await?;
+
+    let mut total_vol = 0;
+    for order_fill in results {
+        total_vol += order_fill.base_quantity;
+    }
+
+    Ok(Json(total_vol))
+}
+
+async fn get_24hr_volume_by_balance_manager_id(
+    Path((pool_id, balance_manager_id)): Path<(String, String)>,
+    State(state): State<PgDeepbookPersistent>,
+) -> Result<Json<Vec<i64>>, DeepBookError> {
+    let connection = &mut state.pool.get().await?;
+    let unix_ts = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as i64;
+    let day_ago = unix_ts - 24 * 60 * 60 * 1000;
+    let results: Vec<OrderFill> = schema::order_fills::table
+        .select(OrderFill::as_select())
+        .filter(schema::order_fills::pool_id.eq(pool_id))
+        .filter(schema::order_fills::onchain_timestamp.gt(day_ago))
+        .load(connection)
+        .await?;
+
+    let mut maker_vol = 0;
+    let mut taker_vol = 0;
+    for order_fill in results {
+        if order_fill.maker_balance_manager_id == balance_manager_id {
+            maker_vol += order_fill.base_quantity;
+        };
+        if order_fill.taker_balance_manager_id == balance_manager_id {
+            taker_vol += order_fill.base_quantity;
+        };
+    }
```

Comment:
i think we should merge this with `get_24hr_volume` and apply another `.filter(schema::order_fills::maker_balance_manager_id.eq(maker_balance_manager_id))`